src/Razor/src/Microsoft.VisualStudio.RazorExtension/RazorPackage.cs

```diff
@@ -52,6 +60,36 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
                 return new RazorLanguageService(breakpointResolver, proximityExpressionResolver, uiThreadOperationExecutor, editorAdaptersFactory, joinableTaskContext.Factory);
             }, promote: true);
+
+            // Add our command handlers for menu (commands must exist in the .vsct file).
+            if (await GetServiceAsync(typeof(IMenuCommandService)) is OleMenuCommandService mcs)
+            {
+                // Create the command for the tool window.
+                var toolwndCommandID = new CommandID(GuidSyntaxVisualizerMenuCmdSet, (int)CmdIDRazorSyntaxVisualizer);
+                var menuToolWin = new MenuCommand(ShowToolWindow, toolwndCommandID);
+                mcs.AddCommand(menuToolWin);
```

Comment:
Does this mean this option will also be there for non-internal users?

Reply:
Yes, this is all visible to everyone. If that's a problem we can make do it just for debug builds, or we could have it as a separate extension published to the VS Marketplace (somehow!?). but I thought given the roslyn syntax visualizer is public, it didn't matter much.

Oh I guess one other option would be to not put the menu item in the View menu, so you'd have to use Ctrl+Q to display it.

Reply:
Answered in the main PR discussion